src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -1290,40 +1301,50 @@ bool SotaUptaneClient::sendMetadataToEcus(const std::vector<Uptane::Target> &tar
       }
 
       /* Root rotation if necessary */
-      rotateSecondaryRoot(Uptane::RepositoryType::Director(), *(sec->second));
-      rotateSecondaryRoot(Uptane::RepositoryType::Image(), *(sec->second));
-      if (!sec->second->putMetadata(target)) {
-        LOG_ERROR << "Sending metadata to " << sec->first << " failed";
-        put_meta_succeed = false;
+      auto result = rotateSecondaryRoot(Uptane::RepositoryType::Director(), *(sec->second));
+      if (!result.isSuccess()) {
+        final_result = result;
+        continue;
+      }
+      result = rotateSecondaryRoot(Uptane::RepositoryType::Image(), *(sec->second));
+      if (!result.isSuccess()) {
+        final_result = result;
+        continue;
+      }
+      result = sec->second->putMetadata(target);
+      if (!result.isSuccess()) {
+        LOG_ERROR << "Sending metadata to " << sec->first << " failed: " << result.result_code << " "
+                  << result.description;
+        final_result = result;
```

Comment:
So if more than one ECU failed, we will report an error only for the last one?

Reply:
Nevermind, didn't notice the TODO on top.

Reply:
Yeah, we should fix that, but since it doesn't affect the protocol or SecondaryInterface, we can follow up with a fix and it won't risk breaking anything.